realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -828,7 +839,47 @@ bool BaseRealSenseNode::shouldPublishCameraInfo(const stream_index_pair& sip)
     return (stream != RS2_STREAM_SAFETY && stream != RS2_STREAM_OCCUPANCY && stream != RS2_STREAM_LABELED_POINT_CLOUD);
 }
 
-void BaseRealSenseNode::publishLabeledPointCloud(rs2::labeled_points lpc, const rclcpp::Time& t){
+void BaseRealSenseNode::publishOccupancyFrame(rs2::frame f, const rclcpp::Time& t)
+{
+    if(!_occupancy_publisher || 0 == _occupancy_publisher->get_subscription_count())
+        return;
+
+    ROS_DEBUG("Publishing Occupancy GridCells Frame");
+
+    // get frame bytes and frame metadata relevant info
+    auto frame_as_uint8_arr = (uint8_t*)f.get_data();
+    auto cols = static_cast<int>(f.get_frame_metadata(RS2_FRAME_METADATA_OCCUPANCY_GRID_COLUMNS));
+    auto rows = static_cast<int>(f.get_frame_metadata(RS2_FRAME_METADATA_OCCUPANCY_GRID_ROWS));
+    auto cell_size = static_cast<float>(f.get_frame_metadata(RS2_FRAME_METADATA_OCCUPANCY_CELL_SIZE) / 100.0f); // convert to meters
+
+    // create GridCells msg and start filling it
+    nav_msgs::msg::GridCells msg;
+    msg.header.stamp = t;
+    msg.header.frame_id = FRAME_ID(OCCUPANCY);
+    msg.cell_width = cell_size;
+    msg.cell_height = cell_size;
+
+    for (auto i = 0; i < cols * rows; ++i)
```

Comment:
Does this means rows and columns represent the cells and not the bytes?
meaning they are in `bit` units?
raws = 8 is 1 byte?

Reply:
rows = grid cell height, cols = grid cell width
added more documentation for this code